tests/cpp-tests/Classes/FileUtilsTest/FileUtilsTest.cpp

```diff
@@ -592,6 +593,127 @@ std::string TestWriteString::subtitle() const
     return "";
 }
 
+class CustomBuffer : public ResizableBuffer {};
+
+struct AlreadyExistsBuffer {};
+
+NS_CC_BEGIN
```

Comment:
Se [in detail section in here](http://en.cppreference.com/w/cpp/language/template_specialization):

> Explicit specialization can only appear at namespace scope, in the same namespace as the primary template (or in the nearest enclosing namespace if the primary is a member template), and it has to appear after the non-specialized template declaration. It is always in the scope of that namespace: